packages/frontend/editor-ui/src/composables/usePushConnection/handlers/executionFinished.ts

```diff
@@ -0,0 +1,496 @@
+import type { ExecutionFinished } from '@n8n/api-types/push/execution';
+import { useUIStore } from '@/stores/ui.store';
+import type { IExecutionResponse } from '@/Interface';
+import { WORKFLOW_SETTINGS_MODAL_KEY } from '@/constants';
+import { getEasyAiWorkflowJson } from '@/utils/easyAiWorkflowUtils';
+import { clearPopupWindowState, hasTrimmedData, hasTrimmedItem } from '@/utils/executionUtils';
+import { useWorkflowsStore } from '@/stores/workflows.store';
+import { useSettingsStore } from '@/stores/settings.store';
+import { useWorkflowHelpers } from '@/composables/useWorkflowHelpers';
+import { useTelemetry } from '@/composables/useTelemetry';
+import { parse } from 'flatted';
+import { useToast } from '@/composables/useToast';
+import type { useRouter } from 'vue-router';
+import { useI18n } from '@/composables/useI18n';
+import { TelemetryHelpers } from 'n8n-workflow';
+import type {
+	IWorkflowBase,
+	NodeError,
+	NodeOperationError,
+	SubworkflowOperationError,
+	ExpressionError,
+	IDataObject,
+	IRunExecutionData,
+} from 'n8n-workflow';
+import { codeNodeEditorEventBus, globalLinkActionsEventBus } from '@/event-bus';
+import { h } from 'vue';
+import NodeExecutionErrorMessage from '@/components/NodeExecutionErrorMessage.vue';
+import { getTriggerNodeServiceName } from '@/utils/nodeTypesUtils';
+import { useExternalHooks } from '@/composables/useExternalHooks';
+import { useNodeHelpers } from '@/composables/useNodeHelpers';
+import { useNodeTypesStore } from '@/stores/nodeTypes.store';
+
+export type SimplifiedExecution = Pick<
+	IExecutionResponse,
+	'workflowId' | 'data' | 'status' | 'startedAt' | 'stoppedAt'
+>;
+
+/**
+ * Handles the 'executionFinished' event, which happens when a workflow execution is finished.
+ */
+export async function executionFinished(
+	{ data }: ExecutionFinished,
+	options: { router: ReturnType<typeof useRouter> },
+) {
+	const workflowsStore = useWorkflowsStore();
+	const uiStore = useUIStore();
+
+	// No workflow is actively running, therefore we ignore this event
+	if (typeof workflowsStore.activeExecutionId === 'undefined') {
+		return;
+	}
+
+	const telemetry = useTelemetry();
+
+	clearPopupWindowState();
```

Comment:
should we move all the code around `formPopupWindow` into `uiStore`. perhaps in another PR as well.

Reply:
cc @autologie

Reply:
Looks like this is about the popup for the form trigger. I agree to move it to uiStore in a separate PR